python/triqs_cthyb/solver.py

```diff
@@ -185,8 +219,14 @@ def solve(self, **params_kw):
                 # Enforce 1/w behavior of G_iw in the tail fit window
                 # and recompute Sigma_iw
                 for name, g in self.G_iw:
-                    tail = np.zeros([2] + list(g.target_shape), dtype=np.complex)
-                    tail[1] = np.eye(g.target_shape[0])
+                    if self.last_solve_parameters["measure_density_matrix"]:
+                        tail = np.zeros([3] + list(g.target_shape), dtype=np.complex)
+                        tail[1] = self.G_moments[name][0]
+                        tail[2] = self.G_moments[name][1]
+                    else:
+                        tail = np.zeros([2] + list(g.target_shape), dtype=np.complex)
+                        tail[1] = np.eye(g.target_shape[0])
```

Comment:
Rename `name` -> `bl` in loop above and
```suggestion
                    if self.last_solve_parameters["measure_density_matrix"]:
                        tail = self.G_moments[bl]
                    else:
                        tail = make_zero_tail(g, 2)
                        tail[1] = np.eye(g.target_shape[0])
```